src/sympc/protocol/aby3/aby3.py

```diff
@@ -169,3 +206,78 @@ def _get_truncation_pair(x: MPCTensor, session: Session) -> Tuple[MPCTensor]:
         r_mpc = MPCTensor(shares=r, session=session, shape=x.shape)
         rPrime_mpc = MPCTensor(shares=rPrime, session=session, shape=x.shape)
         return r_mpc, rPrime_mpc
+
+    @staticmethod
+    def local_decomposition(
+        x: ReplicatedSharedTensor, ring_size: str
+    ) -> List[ReplicatedSharedTensor]:
+        """Performs local decomposition to generate shares of shares.
+
+        Args:
+            x (ReplicatedSharedTensor) : input RSTensor.
+            ring_size (str) : Ring size to generate decomposed shares in.
+
+        Returns:
+            List[ReplicatedSharedTensor]: Decomposed shares in the given ring size.
+
+        Raises:
+            ValueError: If RSTensor does not have session uuid.
+        """
+        if x.session_uuid is None:
+            raise ValueError("Input RSTensor should have session_uuid")
+
+        session = get_session(x.session_uuid)
+        ring_size = int(ring_size)
+        tensor_type = get_type_from_ring(ring_size)
+        rank = session.rank
+        nr_parties = session.nr_parties
+
+        zero = torch.zeros(x.shares[0].shape).type(tensor_type)
+
+        shares = [[zero.clone(), zero.clone()] for i in range(nr_parties)]
+
+        shares[rank][0] = x.shares[0].clone().type(tensor_type)
+
+        shares[(rank + 1) % nr_parties][1] = x.shares[1].clone().type(tensor_type)
+
+        rst_list = []
+        for i in range(nr_parties):
+            rst = x.clone()
+            rst.shares = shares[i]
+            rst.ring_size = ring_size
+            rst_list.append(rst)
+
+        return rst_list
+
+    @staticmethod
+    def bit_injection(x: MPCTensor, session: Session, ring_size: int) -> MPCTensor:
```

Comment:
Add a check to see if shares are a binary and single bit.

Reply:
Yeah, I was delaying that part,  thinking whether to have the sanity check in bit injection or local decomposition, having local decomposition would reduce one round complexity(primary reason), and also local decomposition is only done for binary shares.Shall I include it in local decomposition itself?

Reply:
I would suggest you include both. It would help in debugging. We might use local decomposition somewhere else in the future or someone might add a line of code before local decomposition making the shares non-binary and wouldn't know results are erroneous.

Reply:
Hrishi, I initially thought local decomposition is needed only binary shares, but we require it to be ubiquitous (conversion from any ring size to another) for other ops. I will include the sanity check in bit injection for ring size 2.

Reply:
Done!